File: packages/python-sdk/src/blindfold/regex/detectors/lv/personal_code.py
"""Latvian Personal Code (Personas kods) detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import lv_personal_code_checksum


@register_region("lv")
class PersonalCodeDetector(RegexDetector):
    entity_type = EntityType.LV_PERSONAL_CODE
    score = 0.85
    context_keywords = [
        "personas kods", "personal code", "personas",
    ]
    context_required = True
    context_window = 50

    # 6 digits + optional hyphen + 5 digits
    pattern = re.compile(
        r"\b\d{6}-?\d{5}\b"
    )
    validator = staticmethod(lv_personal_code_checksum)

File: packages/js-sdk/src/regex/detectors/lv/personal-code.ts
// Latvian Personal Code (Personas kods) detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { lvPersonalCodeChecksum } from '../../validators'

class LvPersonalCodeDetector extends RegexDetector {
  entityType = EntityType.LV_PERSONAL_CODE
  score = 0.85
  contextKeywords = [
    'personas kods',
    'personal code',
    'personas',
  ]
  contextRequired = true
  contextWindow = 50

  pattern = /\b\d{6}-?\d{5}\b/g
  validator = lvPersonalCodeChecksum
}

registerRegion('lv', LvPersonalCodeDetector)

export { LvPersonalCodeDetector }
